### PR TITLE
PAS-479 | Refactor application settings

### PR DESCRIPTION
**Pull request by jonashendrickx**
### Ticket

- Closes [PAS-479](https://bitwarden.atlassian.net/browse/PAS-479)

### Description

The final pull request to split all sections up in their own component, including migration of the parent page.

- Some forms were broken as Blazor only accepts one form in the DOM with the same name. This throws an InvalidOperationException.
- Remove backwards compatibility for Razor Pages.
- Vue.js enabled for Blazor
- Allow usage of import maps in Blazor using a simple component.
- Fixes broken confirmation modals on Blazor-based pages.

### Checklist
I did the following to ensure that my changes were tested thoroughly:
- __

I did the following to ensure that my changes do not introduce security vulnerabilities:
- __

[PAS-479]: https://bitwarden.atlassian.net/browse/PAS-479?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/AdminConsole/Components/Pages/App/Settings/SettingsComponents/ApiKeysSection.razor [hunk 1]
```diff
@@ -11,84 +11,89 @@
...
             <button id="create-public-key-btn" class="btn-primary" name="CreateForm.Type" type="submit" value="public">Create Public Key</button>
             <button id="create-secret-key-btn" class="btn-primary ml-4" name="CreateForm.Type" type="submit" value="secret">Create Secret Key</button>
         </EditForm>
-    
-        <Table ColumnHeaders="@(new []{ "Api Key", "Created At", "Type", "Locked", "Scopes", string.Empty })" EmptyMessage="This application has no API Keys or API Secrets.">
-            @if (ApiKeys != null)
-            {
-                @foreach (var apiKey in ApiKeys)
+
+        <ConfirmEditForm Model="ConfirmedSelectedForm" FormName="@(ConfirmedSelectedApiKeyFormName)" name="@ConfirmedSelectedApiKeyFormName" OnSubmit="OnSelectedFormConfirmed">
```
Comment (jonashendrickx) [inline src/AdminConsole/Components/Pages/App/Settings/SettingsComponents/ApiKeysSection.razor:20]: Blazor doesn't like multiple events with the same form name. So we have to wrap the entire table in the form similar to how we manage admins.

File: src/AdminConsole/Components/Pages/App/Settings/Settings.razor [hunk 2]
```diff
@@ -0,0 +1,24 @@
+@page "/app/{AppId}/settings"
+
+@using Passwordless.AdminConsole.Components.Pages.App.Settings.SettingsComponents
+
+@inherits BaseApplicationPage
+
+@inject IDataService DataService
+@inject ICurrentContext CurrentContext
+
+<Page>
```
Comment (jonashendrickx) [inline src/AdminConsole/Components/Pages/App/Settings/Settings.razor:10]: Very easy to split up later when we want to.

File: src/AdminConsole/Components/App.razor [hunk 3]
```diff
@@ -9,16 +9,28 @@
...
     <HeadOutlet/>
     <SecureScript src="lib/passwordless/passwordless.umd.min.js" />
+    <SecureImportMapScript Value="@ImportMap" />
 </head>
 
 <body class="h-full bg-gray-50">
     <Routes/>
     <script src="_framework/blazor.web.js" autostart="false"></script>
     <SecureScript type="text/javascript" src="js/core.js" />
+    <SecureScript type="module" src="js/confirm-form.mjs" />
```
Comment (jonashendrickx) [inline src/AdminConsole/Components/App.razor:19]: Required for confirmation modals, although this required the Vue.js dependency, see line 12.

File: src/AdminConsole/Components/Pages/App/Settings/SettingsComponents/ApiKeysSection.razor [hunk 4]
```diff
@@ -11,84 +11,88 @@
...
```
Comment (jonashendrickx) [inline src/AdminConsole/Components/Pages/App/Settings/SettingsComponents/ApiKeysSection.razor:20]: Required the confirmation modal to be imported into Blazor. See `App.razor`.

File: src/AdminConsole/wwwroot/js/deleteapplication.mjs [hunk 5]
```diff
@@ -1,41 +0,0 @@
-import {ref, computed} from "vue";
```
Comment (jonashendrickx) [inline src/AdminConsole/wwwroot/js/deleteapplication.mjs:1]: Have the code match `Delete organization`.

**Review by jrmccannon — APPROVED** [review]

**Pull request merged**